src/vorta/views/exclude_dialog.py

```diff
@@ -0,0 +1,330 @@
+from PyQt6 import uic
+from PyQt6.QtCore import QModelIndex, QObject, Qt
+from PyQt6.QtGui import QStandardItem, QStandardItemModel
+from PyQt6.QtWidgets import (
+    QAbstractItemView,
+    QApplication,
+    QMenu,
+    QMessageBox,
+    QStyledItemDelegate,
+)
+
+from vorta.i18n import translate
+from vorta.store.models import ExclusionModel
+from vorta.utils import get_asset
+from vorta.views.utils import get_colored_icon, get_exclusion_presets
+
+uifile = get_asset('UI/excludedialog.ui')
+ExcludeDialogUi, ExcludeDialogBase = uic.loadUiType(uifile)
+
+
+class MandatoryInputItemModel(QStandardItemModel):
+    '''
+    A model that prevents the user from adding an empty item to the list.
+    '''
+
+    def __init__(self, profile, parent=None):
+        super().__init__(parent)
+        self.profile = profile
+
+    def setData(self, index: QModelIndex, value, role: int = ...) -> bool:
+        # When a user-added item in edit mode has no text, remove it from the list.
+        if role == Qt.ItemDataRole.EditRole and value == '':
+            self.removeRow(index.row())
+            return True
+        if role == Qt.ItemDataRole.EditRole and ExclusionModel.get_or_none(name=value, profile=self.profile):
+            QMessageBox.critical(
+                self.parent(),
+                'Error',
+                'This exclusion already exists.',
+            )
+            self.removeRow(index.row())
```

Comment:
I think switching the order of the MessageBox and removing the row would be nicer.